src/loaders/anndata-loaders/MatrixZarrLoader.js

```diff
@@ -117,20 +202,95 @@ export default class MatrixZarrLoader extends BaseAnnDataLoader {
     return this.geneNames;
   }
 
-  load() {
+  /**
+   * Class method for loading a gene selection.
+   * @param {Array} selection A list of gene names whose data should be fetched.
+   * @returns {Object} { data } containing an array of gene expression data.
+   */
+  async loadGeneSelection(selection) {
+    const {
+      options: { matrix },
+      store,
+    } = this;
+    const geneNames = await this.loadGeneNames();
+    const indices = selection.map(gene => geneNames.indexOf(gene));
+    const zattrs = await this.getJson(`${matrix}/.zattrs`);
+    const encodingType = zattrs['encoding-type'];
+    let genes;
+    if (encodingType === 'csc_matrix') {
+      const [indptrArr, indexArr, cellXGeneArr] = await this._openSparseArrays();
+      const cellNames = await this.loadCellNames();
+      const { data: cols } = await indptrArr.getRaw(null);
+      // If there is not change in the column indexer, then the data is all zeros
+      genes = await Promise.all(indices.map(async (index) => {
+        const startRowIndex = cols[index];
+        const endRowIndex = cols[index + 1];
+        const isColumnAllZeros = startRowIndex === endRowIndex;
+        const geneData = new Uint8Array(cellNames.length).fill(0);
+        if (isColumnAllZeros) {
+          return geneData;
+        }
+        const { data: rowIndices } = await indexArr.get([slice(startRowIndex, endRowIndex)]);
+        const { data: cellXGeneData } = await cellXGeneArr.get([slice(startRowIndex, endRowIndex)]);
+        for (let rowIndex = 0; rowIndex < rowIndices.length; rowIndex += 1) {
+          geneData[rowIndices[rowIndex]] = cellXGeneData[rowIndex];
+        }
+        return geneData;
+      }));
+    } else if (encodingType === 'csr_matrix') {
+      const cellXGene = await this._loadCSRSparseCellXGene();
+      const cellNames = await this.loadCellNames();
+      genes = indices.map((index) => {
+        const geneData = new Float32Array(cellNames.length).fill(0);
+        for (let i = 0; i < cellNames.length; i += 1) {
+          geneData[i] = cellXGene[i * geneNames.length + index];
+        }
+        return geneData;
+      });
+    } else {
+      if (!this.arr) {
+        this.arr = openArray({ store, path: matrix, mode: 'r' });
+      }
+      genes = await Promise.all(
+        indices.map(index => this.arr.then(z => z.get([null, index])).then(({ data }) => data)),
+      );
+    }
+    return { data: genes.map(i => normalize(i).data), url: null };
```

Comment:
The normalization here happens per-gene which is good for scatterplots (rather than pulling a column out of the normalized expression matrix) but this is called for the heatmap as will when `heatmapFilter` is provided.  We don't want that right?  Normalization for the heatmap should be over the entire "heatmap," not per gene?